pyblish_lite/control.py

```diff
@@ -217,56 +292,74 @@ def on_unexpected_error(error):
             return util.defer(500, on_finished_)
 
         def on_finished_():
-            on_finished()
-            self.was_finished.emit()
+            ''' if `self.is_running` is False then processing was stopped by
+            Stop button so we do not want to execute on_finish method!
+            '''
+            if self.is_running:
+                on_finished()
 
         self.is_running = True
+        self.pair_generator = self._pair_yielder(plugins)
+        self.current_pair = next(self.pair_generator, (None, None))
         util.defer(10, on_next)
 
-    def _iterator(self, plugins, context):
-        """Yield next plug-in and instance to process.
-
-        Arguments:
-            plugins (list): Plug-ins to process
-            context (pyblish.api.Context): Context to process
-
+    def collect(self):
+        """ Iterate and process Collect plugins
```

Comment:
Please format this more nicely; no space at the beginning, use newlines.

Reply:
That's by PEP https://www.python.org/dev/peps/pep-0257/